src/Core/Utility.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @constructor
+ */
+function List () {
+	this.next = this
+	this.prev = this
+	this.length = 0
+}
+List.prototype = {
+	/**
+	 * @param {Element} element
+	 * @param {Element} sibling
+	 * @return {Element}
+	 */
+	insert: function insert (element, sibling) {
+		element.next = sibling
+		element.prev = sibling.prev
+		sibling.prev.next = element
+		sibling.prev = element
+		this.length++
+		
+		return element
+	},
+	/**
+	 * @param {Element} element
+	 * @return {Element}
+	 */
+	remove: function remove (element) {
+		if (this.length < 1) 
+			return
+		
+		element.next.prev = element.prev
+		element.prev.next = element.next
+		this.length--
+		
+		return element
+	},
+	/**
+	 * @return {Element}
+	 */
+	pop: function pop () {
+		return this.remove(this.prev)
+	},
+	/**
+	 * @param {Element} element
+	 * @return {Element}
+	 */
+	push: function push (element) {
+		return this.insert(element, this)
+	},
+	/**
+	 * @param {function} callback
+	 */
+	forEach: function forEach (callback) {
+		for (var i = 0, element = this; i < this.length; ++i)
+			callback.call(this, element = element.next, i)
+	}
+}
+
+/**
+ * @constructor
+ */
+function Hash () {
+	this.hash = ''
```

Comment:
This never changes? So `.has()` only checks if `.set()` has ever been called?

Reply:
Yes.